discoverd/registry.go

```diff
@@ -0,0 +1,154 @@
+package discoverd
+
+import (
+	"encoding/json"
+	"errors"
+	"reflect"
+	"time"
+
+	"github.com/flynn/flannel/Godeps/_workspace/src/github.com/flynn/flynn/discoverd/client"
+	"github.com/flynn/flannel/subnet"
+)
+
+// Network is stored in service metadata and contains both the config and list
+// of subnet leases.
+type Network struct {
+	Config  json.RawMessage   `json:"config"`
+	Subnets map[string][]byte `json:"subnets"`
```

Comment:
This should be a `map[string]*json.RawMessage` otherwise it will be encoded using base64.